hs_file_types/models/timeseries.py

```diff
@@ -486,31 +494,30 @@ def update_sqlite_file(self, user):
         sqlite_file_update(self, sqlite_file_to_update, user)
 
     @classmethod
-    def set_file_type(cls, resource, file_id, user):
-        """
-        Sets a .sqlite or .csv resource file to TimeSeries file type
-        :param resource: an instance of resource type CompositeResource
-        :param file_id: id of the resource file to be set as TimeSeries file type
-        :param user: user who is setting the file type
-        :return:
-        """
-
-        # had to import it here to avoid import loop
-        from hs_core.views.utils import create_folder, remove_folder
+    def check_files_for_aggregation_type(cls, files):
+        """Checks if the specified files can be used to set this aggregation type
+        :param  files: a list of ResourceFile objects
 
-        log = logging.getLogger()
+        :return If the files meet the requirements of this aggregation type, then returns this
+        aggregation class name, otherwise empty string.
+        """
+        if len(files) != 1:
+            # no files or more than 1 file
+            return ""
 
-        # get the resource file
-        res_file = utils.get_resource_file_by_id(resource, file_id)
+        if files[0].extension not in cls.get_allowed_uploaded_file_types():
+            return ""
 
-        if res_file is None or not res_file.exists:
-            raise ValidationError("File not found.")
+        return cls.__name__
 
-        if res_file.extension.lower() not in ('.sqlite', '.csv'):
-            raise ValidationError("Not a valid time series supported file.")
+    @classmethod
+    def set_file_type(cls, resource, user, file_id=None, folder_path=None):
```

Comment:
This method looks awfully similar in each implementation.  I think there is likely room for reuse if we declare supported files for each implementation.

Reply:
There is probably room for more reuse. Have refactored this specific function several times to push as much code as the the base helper functions. I won't have time to further improve this now. If needed I can create an issue to look at this later.